muonDIS/make_nTuple_Tr.py

```diff
@@ -0,0 +1,189 @@
+import ROOT as r
+import os
+import argparse
+import math
+from array import array
+from ShipGeoConfig import ConfigRegistry
+import shipunit as u
+
+r.PyConfig.IgnoreCommandLineOptions = True
+
+# Histogram
+hPmuon = r.TH1F('hPmuon', 'Momentum of muons hitting the SBT;P [GeV];Entries', 400, 0., 400)
+
+# Argument parser
+parser = argparse.ArgumentParser(description='Script to collect muons hitting the Tracking Station')
+parser.add_argument('-o', '--outputfile', default='muonsProduction_wsoft_Tr.root')
+parser.add_argument("-p", dest="path", help="path to muon background files", required=True)
+parser.add_argument('-test', dest="testing_code", help="Run Test", action="store_true")
+args = parser.parse_args()
+
+# Variables
+ev = 0
+evs = [0, 0, 0]
+processed_events = set()
+
+
+if args.testing_code: 
+    print("test code, output file name overwritten as: muonsProduction_wsoft_Tr_test.root")
+    args.outputfile='muonsProduction_wsoft_Tr_test.root'
+    selectedmuons='SelectedMuonsTr_test.txt'
+else:
+    selectedmuons='SelectedMuonsTr.txt'
+
+# Output file and tree setup
+output_file = r.TFile(args.outputfile, 'recreate')
+output_tree = r.TTree('MuonAndSoftInteractions', 'Muon information and soft interaction tracks')
+
+imuondata = r.TVectorD(9)  # 9 values: pid, px, py, pz, x, y, z, weight,time_of_hit
+output_tree.Branch('imuondata', imuondata)
+
+track_array = r.TObjArray()
+output_tree.Branch('tracks', track_array)
+
+muon_vetoPoints    = r.TClonesArray("vetoPoint")
+output_tree.Branch('muon_vetoPoints', muon_vetoPoints)
+
+path = args.path
+
+fsel = open(selectedmuons, "w")
+
+# Process each input file
+for inputFile in os.listdir(path):
+    print(inputFile)
+    if args.testing_code and evs[2]>=5:
+        break
+
+    f = r.TFile.Open(os.path.join(path, inputFile), 'read')
+    try:
+        tree = f.cbmsim
+    except Exception as e:
+        print(f"Error: {e}")
+        continue
+
+    for event in tree:
+        
+        ev += 1
+        
+        #saving soft tracks
+        track_array.Clear()
+        muon_id = next((itrk for itrk in range(event.MCTrack.GetEntries()) if abs(event.MCTrack[itrk].GetPdgCode()) == 13), None)
+        for track in event.MCTrack:
+            if track.GetMotherId()==muon_id and (not track.GetProcName().Data() =='Muon nuclear interaction' ):
+                track_array.Add(track)
+                
+        #saving the incoming muon's veto response
+        index=0
+        muon_vetoPoints.Clear()
+        for hit in event.vetoPoint:
+            
+            detID = hit.GetDetectorID()
+            pid = hit.PdgCode()
+            if 1000 < detID < 999999 and abs(pid) == 13:
+                if muon_vetoPoints.GetSize() == index:
+                    muon_vetoPoints.Expand(index+1)
+                muon_vetoPoints[index] = hit
```

Comment:
This should be adapted for ROOT 6.32 as done in https://github.com/ShipSoft/FairShip/pull/559